uvloop/loop.pyx

```diff
@@ -89,6 +89,22 @@ cdef inline socket_dec_io_ref(sock):
         sock._decref_socketios()
 
 
+cdef inline run_in_context(context, method):
+    # This method is internally used to workaround a reference issue that in
+    # certain circumstances, inlined context.run() will not hold a reference to
+    # the given method instance, which - if deallocated - will cause segault.
+    # See also: edgedb/edgedb#2222
+    return context.run(method)
```

Comment:
```suggestion
    INCREF(method)
    try:
        return context.run(method)
    finally:
        DECREF(method)
```

(this is what I meant, just in case)